### PR TITLE
Notebook on web crawling

### PR DESCRIPTION
**Pull request by WilliamEspegren**
## Why are these changes needed?

Lacking web crawling examples in both notebooks and the docs/Example.md

## Checks
Not any applicable right?
- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

I referred to a previous [pull request](https://github.com/microsoft/autogen/pull/2720/files) for inspiration while creating this notebook example for Spider. This PR provided a helpful reference for structure and content, which I have adapted to fit the specific requirements of Spider.

Twitter handle: [@WilliamEspegren](https://x.com/WilliamEspegren)

### CODE REVIEW

**Comment by WilliamEspegren** [comment]
@microsoft-github-policy-service agree

**Comment by sonichi** [comment]
Thanks. Would you like the notebook to be rendered on the [website](https://microsoft.github.io/autogen/docs/notebooks)? If so, https://github.com/microsoft/autogen/blob/main/notebook/contributing.md#how-to-get-a-notebook-displayed-on-the-website is the guide.

**Comment by WilliamEspegren** [comment]
> Thanks. Would you like the notebook to be rendered on the [website](https://microsoft.github.io/autogen/docs/notebooks)? If so, https://github.com/microsoft/autogen/blob/main/notebook/contributing.md#how-to-get-a-notebook-displayed-on-the-website is the guide.

Yes please! I will look into this

**Comment by WilliamEspegren** [comment]
@sonichi This notebook will be rendered on the website right?

**Comment by sonichi** [comment]
> @sonichi This notebook will be rendered on the website right?

Some metadata needs to be added for it to render. It's OK to do it in a separate PR if you like.

**Comment by WilliamEspegren** [comment]
@sonichi Sorry for that, now it is formatted correctly and passes the checks 

image

**Comment by WilliamEspegren** [comment]
> @sonichi Sorry for that, now it is formatted correctly and passes the checks 
> 
> 
> 
> image

Ready to merge :)

**Comment by WilliamEspegren** [comment]
@sonichi Ready to merge :)

**Comment by WilliamEspegren** [comment]
@sonichi Still ready to merge :100:

**Comment by WilliamEspegren** [comment]
@sonichi Should I change this PR in any way?

File: notebook/agentchat_webcrawling_with_spider.ipynb [hunk 1]
```diff
@@ -0,0 +1,426 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Web Scraping using Spider\n",
```
Comment (ekzhu) [inline notebook/agentchat_webcrawling_with_spider.ipynb:7]:
```suggestion
    "# Web Scraping using Spider API\n",
```

File: notebook/agentchat_webcrawling_with_spider.ipynb [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+    "\n",
+    "This notebook shows how to use the fastest open \n",
```
Comment (ekzhu) [inline notebook/agentchat_webcrawling_with_spider.ipynb:9]:
```suggestion
    "This notebook shows how to use the open \n",
```

File: notebook/agentchat_webcrawling_with_spider.ipynb [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+    "source web crawler together with AutoGen agents."
```
Comment (ekzhu) [inline notebook/agentchat_webcrawling_with_spider.ipynb:10]:
```suggestion
    "source [Spider](https://spider-analyzer.io/) web crawler together with AutoGen agents."
```

File: website/docs/Examples.md [hunk 4]
```diff
@@ -55,6 +55,7 @@ Links to notebook examples:
 - Browse the Web with Agents - [View Notebook](https://github.com/microsoft/autogen/blob/main/notebook/agentchat_surfer.ipynb)
 - **SQL**: Natural Language Text to SQL Query using the [Spider](https://yale-lily.github.io/spider) Text-to-SQL Benchmark - [View Notebook](https://github.com/microsoft/autogen/blob/main/notebook/agentchat_sql_spider.ipynb)
 - **Web Scraping**: Web Scraping with Apify - [View Notebook](/docs/notebooks/agentchat_webscraping_with_apify)
+- **Web Crawling**: Crawl entire domain with Spider - [View Notebook](/docs/notebooks/agentchat_webcrawling_with_spider)
```
Comment (ekzhu) [inline website/docs/Examples.md:58]:
```suggestion
- **Web Crawling**: Crawl entire domain with Spider API - [View Notebook](/docs/notebooks/agentchat_webcrawling_with_spider)
```

**Review by ekzhu — COMMENTED** [review]
Made some small edits.

File: notebook/agentchat_webcrawling_with_spider.ipynb (hunk 3 above)
Comment (WilliamEspegren) [reply notebook/agentchat_webcrawling_with_spider.ipynb:10]: spiders website is https://spider.cloud

File: notebook/agentchat_webcrawling_with_spider.ipynb (hunk 2 above)
Comment (WilliamEspegren) [reply notebook/agentchat_webcrawling_with_spider.ipynb:9]: this shows that it is the fastest: https://github.com/spider-rs/spider/blob/main/benches/BENCHMARKS.md

Comment (ekzhu) [reply notebook/agentchat_webcrawling_with_spider.ipynb:9]: Congrats for being the fastest, but would you come back and update our website if someone else tops it?

File: notebook/agentchat_webcrawling_with_spider.ipynb (hunk 3 above)
Comment (ekzhu) [reply notebook/agentchat_webcrawling_with_spider.ipynb:10]: Sorry. Can you update the link?

**Comment by rysweet** [comment]
This PR is against AutoGen 0.2. AutoGen 0.2 has been moved to the 0.2 branch. Please rebase your PR on the 0.2 branch or update it to work with the new AutoGen 0.4 that is now in main.

**Comment by rysweet** [comment]
@ekzhu - looks like OP updated this one - can you check out the docs file conflict?

File: notebook/agentchat_webcrawling_with_spider.ipynb (hunk 2 above)
Comment (ekzhu) [reply notebook/agentchat_webcrawling_with_spider.ipynb:9]: This is not resolved. I don't think we can claim this in our documentation.

**Comment by ekzhu** [comment]
> @ekzhu - looks like OP updated this one - can you check out the docs file conflict?

Left a comment on an unaddressed point

**Comment by WilliamEspegren** [comment]
@ekzhu sorry, fixed now :)

**Review by ekzhu — APPROVED** [review]
Merging this for now but be aware of the upcoming new extension policy for v0.4 (currently preview) onward.

**Pull request merged**